format/Schema.fbs

```diff
@@ -187,6 +187,103 @@ table Interval {
   unit: IntervalUnit;
 }
 
+
+/// ----------------------------------------------------------------------
+/// user defined key value pairs to add custom metadata to arrow
+/// key namespacing is the responsibility of the user
+
+table KeyValue {
+  key: string;
+  value: string;
+}
+
+/// ------------------------------------------------------------------------
+/// Types used to represent a packed c-like structs.  These are experimental
+/// and maybe subject to change.
+/// IMPORTANT: COMPATIBILITY MAY BE BROKEN IN FUTURE release.
+/// They are not required to be implemented.
+
+/// Similar to a c-union, takes the byte-width
+/// of its longest sub-type.  To record type of union
+/// use the "tagged union" pattern.
+table PackedUnion {
+}
+
+/// A binary string of a maximum size.  Fields are layed out as
+/// [length][UTF8 bytes].  The number of bytes used is maxSize +
+/// lengthType.bitWidth/8.
+table PackedBinary {
+  /// Maximum number of bytes (not characters) used.  This
+  /// amount of bytes will always be used.
+  maxSize: int;
+
+  /// If not set all binary data are of maxSize.  If set use specified
```

Comment:
What does "not set" mean?

Reply:
Since this is a  flabtbuffer Table type, I thought one option was for it to be nullable (i.e. not set).  Is this not the case.

Reply:
Ah, I don't know. Sorry :-/
@wesm

Reply:
I added some examples.

My intent was this is always fixed size.  Like I said above I think this is likely too complicated a type to start out with.   I was originally imagining for this and PakcedUtf, that they correspond to something like CHAR(N) in databases 
(but I think char generally is in terms of characters and not bytes).

 I think we might want something like this if we ever want to store generic packed tuples, but it doesn't quite fit into the c-model (users could build this themselve).